test/Dialect/FIRRTL/annotations.fir

```diff
@@ -1108,3 +1108,54 @@ circuit Sub : %[[{
 ; CHECK-SAME: #firrtl<"subAnno<fieldID = 1, {class = \22sifive.enterprise.grandcentral.SignalDriverAnnotation\22, dir = \22sink\22, id = [[ID]] : i64, peer = \22~Top|Foo>dataIn.a.b.c\22, side = \22local\22, targetId = 6 : i64}>">
 ; CHECK-SAME: #firrtl<"subAnno<fieldID = 2, {class = \22sifive.enterprise.grandcentral.SignalDriverAnnotation\22, dir = \22sink\22, id = [[ID]] : i64, peer = \22~Top|Foo>dataIn.d\22, side = \22local\22, targetId = 7 : i64}>">
 ; CHECK-SAME: #firrtl<"subAnno<fieldID = 3, {class = \22sifive.enterprise.grandcentral.SignalDriverAnnotation\22, dir = \22sink\22, id = [[ID]] : i64, peer = \22~Top|Foo>dataIn.e\22, side = \22local\22, targetId = 8 : i64}>">
+
+; // -----
+
+; Grand Central's ModuleReplacementAnnotation is properly scattered to the circuit
+; and the targeted operations.
+
+
+circuit Top : %[[{
+  "class": "sifive.enterprise.grandcentral.ModuleReplacementAnnotation",
+  "targets": [
+    "~Top|Top/child:Child",
+    "~Top|Top/childWrapper:ChildWrapper/child:Child"
+  ],
+  "circuit": "",
+  "annotations": [],
```

Comment:
It would be good to populate these for this test and make sure that they are copied over correctly.